### PR TITLE
Add ability to cache court dates in the django postgres database (again)

### PR DESCRIPTION
**Pull request by timcowlishaw**
## Changes in this PR:

Re-applying this commit after reverting it -before merging we will need to make sure that migrations are run on deploy.

This stems from a discussion with Nick about ways of improving the current situation where we need to manually update the court metadata in the utils library every time a new judgment is published for a particular court in 2023.

Nick propose a cron job that populates the database with the start and end years for each court's current ingested range  on a daily or weekly basis - this handles the persistence layer in the app and the logic to choose the correct date range when displaying a court.

## Trello card / Rollbar error (etc)

https://trello.com/c/NElb5lZV/324-bau-updating-court-dates-from-marklogic

### CODE REVIEW

File: judgments/templatetags/court_utils.py
Comment (anthonyhashemi) [inline judgments/templatetags/court_utils.py:22]:
```suggestion
        start_year = court_dates.start_year
```

File: judgments/templatetags/court_utils.py
Comment (anthonyhashemi) [inline judgments/templatetags/court_utils.py:23]:
```suggestion
        end_year = court_dates.end_year
```

File: judgments/models.py
Comment (anthonyhashemi) [inline judgments/models.py:124]: not a concern for now - but do we envisage adding other court info to the db?

if so, we may consider generalising this class to something like `Court`.

Comment (timcowlishaw) [reply judgments/models.py:124]: Aha, so, i started a spike for moving all the metadata into the DB and it complicated things considerably, so decided to keep this PR more specific. Agree that it's probably a good medium-term aim though!

**Comment by timcowlishaw** [comment]
Folks, I'm gonna mark this ready for review before leaving but please *DO NOT MERGE* until we're happy the migrations will run on deploy!

**Review by dragon-dxw — APPROVED** [review]

**Review by jacksonj04 — APPROVED** [review]

**Pull request merged**
